File: 10 Days of Statistics/Python/Day1 - Interquartile Range.py
#!/bin/python3

import math
import os
import random
import re
import sys
from statistics import median

#
# Complete the 'interQuartile' function below.
#
# The function accepts following parameters:
#  1. INTEGER_ARRAY values
#  2. INTEGER_ARRAY freqs
#

def interQuartile(values, freqs):
    arr = []    
    for i in range(n):
        arr += [values[i]] * freq[i]        

    arr.sort()
    
    middle = len(arr) // 2    
    if n % 2 != 0:
        lm = median(arr[:middle])
        um = median(arr[middle+1:])
    else:
        lm = median(arr[:middle])
        um = median(arr[middle:])
    
    fin = round(float(um - lm), 1)
    
    print(fin)
    
    # Print your answer to 1 decimal place within this function

if __name__ == '__main__':
    n = int(input().strip())

    val = list(map(int, input().rstrip().split()))

    freq = list(map(int, input().rstrip().split()))

    interQuartile(val, freq)

File: 10 Days of Statistics/CS/Day1 - Interquartile Range.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{
    public static double median(List<int> arr)
    {
        arr.Sort();
        int mid = arr.Count / 2;
        if (arr.Count % 2 == 1)
            return (double)arr[mid];
        else
            return ((double)arr[mid] + (double)arr[mid - 1]) / 2;
    }

    /*
     * Complete the 'interQuartile' function below.
     *
     * The function accepts following parameters:
     *  1. INTEGER_ARRAY values
     *  2. INTEGER_ARRAY freqs
     */

    public static void interQuartile(List<int> values, List<int> freqs)
    {
        List<int> arr = new List<int>();
        for (int i = 0; i < values.Count; i++)
            for (int j = 0; j < freqs[i]; j++)
                arr.Add(values[i]);

        arr.Sort();

        int middle = arr.Count / 2;
        double lm = median(arr.GetRange(0, middle));
        double um = median(arr.GetRange((arr.Count % 2 != 0) ? middle + 1 : middle, middle));

        double fin = Math.Round(um - lm, 1);

        Console.WriteLine(fin.ToString(".0"));
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> val = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(valTemp => Convert.ToInt32(valTemp)).ToList();

        List<int> freq = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(freqTemp => Convert.ToInt32(freqTemp)).ToList();

        Result.interQuartile(val, freq);
    }
}
